docs/troubleshooting-guide/sql-logic.md

```diff
@@ -0,0 +1,261 @@
+
+**Further reading**
+
+- [What is a schema?][schema-def]
+- [SQL join types][types-of-joins]
+
+#### My result has missing rows
+
+1. Check if your source tables or query results have missing rows. 
+2. Check the [table below](#how-joins-filter-out-unmatched-rows) to see if you are missing rows because of your join type.
+3. Check your join conditions in the `ON` clause. For example:
+    ```sql
+    # The join condition below will remove:
+    # All rows from table A where key_a = "foo"
+    # All rows from table B where key_b = "foo"
+
+    SELECT *
+    FROM table_a
+    JOIN table_b
+    ON key_a = key_b
+    AND key_b <> "foo"
+    ```
+4. Check if your `WHERE` clause is interacting with your `JOIN` clause. For example:
+    ```sql
+    # The WHERE clause below will remove:
+    # All rows from table A where key_a = "foo"
+    # All rows from table B where key_b = "foo"
+
+    SELECT *
+    FROM table_a
+    JOIN table_b
+    WHERE key_b <> "foo"
+    ```
+5. If you want to *add* rows to your query result to fill in data that is empty, zero, or `NULL`, you need to start your JOINs with a table or column that has all the rows you want. Ask your database admin if there’s a table you can use for this, or can create a temporary column using `GENERATE_SERIES`.
+    ```sql
+    # This result has one row for every day with at least one transaction.
+    SELECT 
+    o.date
+    , SUM(p.price) AS total_sales
+    FROM orders o
+    LEFT JOIN products p
+    ON o.product_id = p.id
+    GROUP BY 1
+    ;
+
+    # This result has one row for every day, including days with 0 transactions.
+    WITH date_series AS (
+    GENERATE_SERIES(...)
+    )
+
+    SELECT 
+    d.date
+    , SUM(p.price) AS total_sales
+    FROM date_series d
+    LEFT JOIN orders o
+        ON d.date = o.order_date
+    LEFT JOIN products p
+        ON o.product_id = p.id
+    GROUP BY 1
+    ;
+    ```
+6. If you’re getting an error message, go to [Troubleshooting error messages][troubleshooting-error-messages].
+7. If you're still stuck, search or ask the [Metabase community][discourse].
+
+#### How joins filter out unmatched rows
+| Join type | Rows that are filtered out |
+| --- | --- |
+| Table A INNER JOIN Table B | You will lose rows from both table A and table B if the join condition is not met. |
+| Table A LEFT JOIN Table B | You will lose rows from table B if the join condition is not met. |
+| Table B LEFT JOIN Table A | You will lose rows from table A if the join condition is not met. |
+| Table A OUTER JOIN Table B | You will lose rows from both table A and table B if the join condition is met. |
+| Table A FULL JOIN Table B | This join does not remove any rows from table A or table B. |
+
+**Explanation**
+
+Rows can get removed by accident when data gets refreshed in upstream systems or ETL jobs. 
```

Comment:
Link to ETL learn article? https://www.metabase.com/learn/analytics/etl-landscape